### PR TITLE
Monaco goes collaboration

### LINKED ISSUE #41: Integrate useCollaboration hook with the app

### PR DESCRIPTION
**Pull request by katzper01**
- Integration of #37 with the app
- two smaller things I had locally and didn't want to make separate PRs for them :P: 
    - useKeyDown hook and `Ctrl-S` hotkey to trigger compilation
    - `LoadingOverlay` component. You can wrap components with it to display a loading indicator over them. Used for editor when compilation is in LOADING state

Resolves #41

### CODE REVIEW

**Review by hubzie — APPROVED** [review]

File: src/util/poly/poly.ts
[the remark this answers is not in the document]
Comment (katzper01) [reply src/util/poly/poly.ts:1]: https://pl.m.wikipedia.org/wiki/Polyfill

File: src/components/editor/Editor.tsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+  const [ delimiterX, setDelimiterX ] = useState(0.5);
+  const latexTextAreaWidth = `${100*delimiterX}%`;
+  const pdfViewerWidth = `calc(${100*(1 - delimiterX)}% - ${EDITOR_DELIMITER_WIDTH}px)`;
+  const pdfViewerWidthValue = (width ?? 0) * (1 - delimiterX) - EDITOR_DELIMITER_WIDTH;
+
+  const onDrag = (x: number) => {
+    setDelimiterX(x);
+  };
+
+  const getPreviewComponent = (): ReactNode => {
```
[the remark this answers is not in the document]
Comment (katzper01) [reply src/components/editor/Editor.tsx]: yess

File: src/components/editor/Editor.tsx [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        }}
+        viewMinPercentageWorkspaceWidth={EDITOR_MIN_PERCENTAGE_WORKSPACE_WIDTH}
+        delimiterWidth={EDITOR_DELIMITER_WIDTH}
+        onDrag={onDrag}
+        testId='delimiter'/>
+      <div 
+        className={styles.viewer}
+        style={{ width: pdfViewerWidth }}
+      >
+        { getPreviewComponent() }
```
[the remark this answers is not in the document]
Comment (katzper01) [reply src/components/editor/Editor.tsx]: yes

**Review by hubzie — COMMENTED** [review]
Looks good so far

**Pull request merged**